src/libstd/keyword_docs.rs

```diff
@@ -495,9 +518,16 @@ mod if_keyword { }
 /// [book2]: ../book/ch10-02-traits.html#returning-types-that-implement-traits
 mod impl_keyword { }
 
+#[doc(keyword = "in")]
+//
+/// Iterate over a series of values with [`for`].
+///
+/// [`for`]: keyword.for.html
+mod in_keyword { }
+
 #[doc(keyword = "let")]
 //
-/// The variable binding keyword.
+/// Bind a value to a variable.
```

Comment:
`let` is used for general pattern matching, which may include binding variables but may also not.

Reply:
I would say that binding a value to (one or more) variable(s) is by far the most common of usage of `let`. For bare `let` I can think of `let _ = ...` and irrefutable patterns like `let UnitStruct = v` as exceptions. `let` without binding is more useful in situations like `if let Enum::UnitVariant = v` but is pretty rare in my personal experience.

Reply:
Well you can say something like `let Wrapping(my_binding) = expr;` which both does irrefutable pattern matching and binds a value. Pattern matching and "irrefutable" are also terminologies we already use.

Possibly, you could go into the common case in the first line and then talk about the general case in later paragraphs or invert the order. However, I think the info about the general case should be included somehow.

Reply:
How about "Bind to a value by pattern matching"?

Reply:
In that case:
```suggestion
/// Bind values by pattern matching.
```
(note the plural)

Reply:
The right hand side of a let statement is an expression, the result of which is a single value. That value can be a composite (e.g. `(1, 2)`),  but I believe a `let` statement only operates on one value at a time. I could agree with "create zero or more bindings to a value by pattern matching" but I think it's more verbose than necessary.

Reply:
I see what you're saying... Upon re-reading the original, I think it's servicable for now so I'll resolve this.